apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/ResponseFieldSpec.kt

```diff
@@ -0,0 +1,293 @@
+package com.apollographql.apollo.compiler
+
+import com.apollographql.apollo.api.ResponseField
+import com.apollographql.apollo.api.ResponseReader
+import com.apollographql.apollo.compiler.ir.CodeGenerationContext
+import com.apollographql.apollo.compiler.ir.Field
+import com.squareup.javapoet.*
+import java.io.IOException
+import java.util.*
+import javax.lang.model.element.Modifier
+
+class ResponseFieldSpec(
+    val irField: Field,
+    val fieldSpec: FieldSpec,
+    val responseFieldType: ResponseField.Type,
+    val typeConditions: List<String> = emptyList(),
+    val context: CodeGenerationContext
+) {
+
+  fun factoryCode(): CodeBlock {
+    val factoryMethod = FACTORY_METHODS[responseFieldType]!!
```

Comment:
we can't have `factoryMethod = null`, otherwise this is our error (we forgot to add mapping response field type to appropriate factory method)  and we need to crash here.